src/main/java/org/sagebionetworks/web/client/widget/entity/EntityBadge.java

```diff
@@ -73,6 +82,13 @@ public void invoke() {
 		};
 		isConfigured = false;
 		isAttached = false;
+		fileDownloadButton.setSize(ButtonSize.EXTRA_SMALL);
+		fileDownloadButton.setEntityUpdatedHandler(new EntityUpdatedHandler() {
+			@Override
+			public void onPersistSuccess(EntityUpdatedEvent event) {
+				getEntityBundle();
+			}
+		});
```

Comment:
what happens if the user does not have DOWNLOAD permission on the entity?
what happens for anonymous?